### ISSUE TITLE
[iOS] No matching member function for call to 'invokeAsync' [React Native Version --> 0.75.1]

### ISSUE THREAD

**Issue by aviral1518** [realm/realm-js#6846]
### How frequently does the bug occur?

Always

### Description

I am getting these issues while building an iOS application. I am using `react-native: 0.75.1`. 

<img width="925" alt="image" src="https://github.com/user-attachments/assets/747941e0-1900-4d9a-bfc3-839a89e27139">

### Stacktrace & log output

```shell
Showing Recent Issues
/Users/aviralkhandelwal/Desktop/Projects/efoodi-react-native/node_modules/realm/binding/jsi/react_scheduler.cpp:59:28: No matching member function for call to 'invokeAsync'


Showing Recent Issues
/Users/aviralkhandelwal/Desktop/Projects/efoodi-react-native/ios/Pods/Headers/Public/React-callinvoker/ReactCommon/CallInvoker.h:30:16: Candidate function not viable: no known conversion from '(lambda at /Users/aviralkhandelwal/Desktop/Projects/efoodi-react-native/node_modules/realm/binding/jsi/react_scheduler.cpp:61:65)' to 'CallFunc' (aka 'function<void (jsi::Runtime &)>') for 2nd argument


Showing Recent Issues
/Users/aviralkhandelwal/Desktop/Projects/efoodi-react-native/ios/Pods/Headers/Public/React-callinvoker/ReactCommon/CallInvoker.h:29:16: Candidate function not viable: requires single argument 'func', but 2 arguments were provided


Showing Recent Issues
/Users/aviralkhandelwal/Desktop/Projects/efoodi-react-native/ios/Pods/Headers/Public/React-callinvoker/ReactCommon/CallInvoker.h:40:16: Candidate function not viable: requires single argument 'func', but 2 arguments were provided
```

### Can you reproduce the bug?

Always

### Reproduction Steps

1. `yarn add realm`
2. `yarn add @realm/react`
3.  On iOS directory run `pod install`
4. Make build using Xcode
5. This issue comes while building the application

<img width="450" alt="image" src="https://github.com/user-attachments/assets/19f9dbc6-a4f7-4e1c-894d-0c1d22fcb6c9">

### Version

0.10.0

### What services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

Platform --> iOS, XCode Version --> 15.4, React Native Version --> 0.75.1, Realm Version --> 12.13.0, realm/react Version --> 0.10.0

### Build environment

{
  "name": "efoodi",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "lint": "eslint .",
    "start": "react-native start",
    "test": "jest",
    "prepare": "patch-package",
    "postinstall": "patch-package && npx jetify"
  },
  "lint-staged": {
    "*.{js,jsx,ts,tsx}": [
      "prettier --write",
      "eslint --fix"
    ]
  },
  "config": {
    "commitizen": {
      "path": "cz-conventional-changelog"
    }
  },
  "dependencies": {
    "@react-native-async-storage/async-storage": "^1.24.0",
    "@react-native-community/eslint-config": "^3.2.0",
    "@react-native-masked-view/masked-view": "^0.3.1",
    "@react-navigation/native": "^6.1.18",
    "@react-navigation/stack": "^6.4.1",
    "@realm/react": "^0.10.0",
    "@reduxjs/toolkit": "^2.2.7",
    "i18next": "^23.12.3",
    "jetifier": "^2.0.0",
    "patch-package": "^8.0.0",
    "postinstall": "^0.10.3",
    "react": "18.3.1",
    "react-i18next": "^15.0.1",
    "react-native": "0.75.1",
    "react-native-fast-image": "^8.6.3",
    "react-native-geolocation-service": "^5.3.1",
    "react-native-gesture-handler": "^2.18.1",
    "react-native-linear-gradient": "^2.8.3",
    "react-native-paper": "^5.12.5",
    "react-native-permissions": "^4.1.5",
    "react-native-restart": "^0.0.27",
    "react-native-safe-area-context": "^4.10.8",
    "react-native-screens": "^3.34.0",
    "react-native-vector-icons": "^10.1.0",
    "react-redux": "^9.1.2",
    "realm": "^12.13.0",
    "redux": "^5.0.1",
    "redux-logger": "^3.0.6"
  },
  "devDependencies": {
    "@babel/core": "^7.25.2",
    "@babel/eslint-parser": "^7.25.1",
    "@babel/preset-env": "^7.20.0",
    "@babel/runtime": "^7.20.0",
    "@commitlint/cli": "^19.4.0",
    "@commitlint/config-conventional": "^19.2.2",
    "@eslint/js": "^9.9.0",
    "@react-native/babel-preset": "0.75.1",
    "@react-native/eslint-config": "0.75.1",
    "@react-native/metro-config": "0.75.1",
    "@react-native/typescript-config": "0.75.1",
    "@types/react": "^18.2.6",
    "@types/react-test-renderer": "^18.0.0",
    "@typescript-eslint/eslint-plugin": "^8.1.0",
    "@typescript-eslint/parser": "^8.1.0",
    "babel-jest": "^29.6.3",
    "commitizen": "^4.3.0",
    "cz-conventional-changelog": "^3.3.0",
    "eslint": "^9.9.0",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-airbnb-typescript": "^18.0.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-import-resolver-typescript": "^3.6.1",
    "eslint-plugin-import": "^2.29.1",
    "eslint-plugin-jsx-a11y": "^6.9.0",
    "eslint-plugin-n": "^17.10.2",
    "eslint-plugin-prettier": "^5.2.1",
    "eslint-plugin-promise": "^7.1.0",
    "eslint-plugin-react": "^7.35.0",
    "eslint-plugin-react-hooks": "^4.6.2",
    "eslint-plugin-react-native": "^4.1.0",
    "globals": "^15.9.0",
    "husky": "^9.1.4",
    "jest": "^29.6.3",
    "lint-staged": "^15.2.9",
    "prettier": "^3.3.3",
    "react-test-renderer": "18.3.1",
    "reactotron-react-native": "^5.1.7",
    "reactotron-redux": "^3.1.9",
    "typescript": "5.0.4",
    "typescript-eslint": "^8.1.0"
  },
  "engines": {
    "node": ">=18"
  },
  "packageManager": "yarn@3.6.4"
}

### Cocoapods version

1.15.2

**Comment by nthtrung09it** [comment]
same issue :(

**Comment by nthtrung09it** [comment]
@aviral1518 You can try to modify the react_scheduler.cpp to see if it works:
![CleanShot 2024-08-17 at 20 48 26@2x](https://github.com/user-attachments/assets/820ee713-5b3e-4cbd-9657-8d22ec15ed1b)

**Comment by aviral1518** [comment]
> @aviral1518 You can try to modify the react_scheduler.cpp to see if it works: ![CleanShot 2024-08-17 at 20 48 26@2x](https://private-user-images.githubusercontent.com/7716509/358849107-820ee713-5b3e-4cbd-9657-8d22ec15ed1b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5MTA5MzcsIm5iZiI6MTcyMzkxMDYzNywicGF0aCI6Ii83NzE2NTA5LzM1ODg0OTEwNy04MjBlZTcxMy01YjNlLTRjYmQtOTY1Ny04ZDIyZWMxNWVkMWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTdUMTYwMzU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjYzMTYzMWM5MzVhNWRmOTMxNGE5OWIxODJjNjlmZjRkYzFkY2RhMWI1NzMzNmE2NTAwOTk2MzdkNmJhMjQyNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.PsHw4VcL5ufJq9nDtLAFR0TxruOJeolB9aWGeOQdtV0)

@nthtrung09it I can build it by making the changes that you have shared. It saved me a lot of time. Thank you so much. 😊

BTW this same issue I am also facing while making Android Build. Do you have any idea how to fix that, too?

**Comment by nthtrung09it** [comment]
> BTW this same issue I am also facing while making Android Build. Do you have any idea how to fix that, too?

@aviral1518  With the same changes I could build Android too.

**Comment by aviral1518** [comment]
> > BTW this same issue I am also facing while making Android Build. Do you have any idea how to fix that, too?
> 
> @aviral1518 With the same changes I could build Android too.

@nthtrung09it in Android, which file do I need to make changes and where ?

**Comment by nthtrung09it** [comment]
> @nthtrung09it in Android, which file do I need to make changes and where ?

@aviral1518 I only changed the file in the image, and it works for Android and iOS.
https://github.com/realm/realm-js/issues/6846#issuecomment-2294865629

**Comment by kraenhansen** [comment]
This is a symptom of the same underlying issue reported in #6845.

**Comment by kraenhansen** [comment]
A fix was release with [v12.13.1](https://github.com/realm/realm-js/releases/tag/v12.13.1) - please take it for a spin and let us know how it goes 👍

**Comment by aviral1518** [comment]
> A fix was release with [v12.13.1](https://github.com/realm/realm-js/releases/tag/v12.13.1) - please take it for a spin and let us know how it goes 👍

@kraenhansen working fine now. Thanks
